src/parsers/ddi_parser.jl

```diff
@@ -0,0 +1,189 @@
+
+@kwdef mutable struct DDIVariable
+    name::String = ""
+    position_start::Int64 = 9999
+    position_end::Int64 = 9999
+    position_width::Int64 = 9999 
+    labl::String = ""
+    txt::String = ""
+    dcml::Int64 = 9999
+    vartype::String = ""
+end
+
+@kwdef mutable struct DDIInfo
+    filepath::String
+    conditions::String = ""
+    citation::String = ""
+    ipums_project::String = ""
+    extract_notes::String = ""
+    extract_date::String = ""
+    variable_info::Vector{DDIVariable} = DDIVariable[]
+    _xml_doc::EzXML.Document = EzXML.XMLDocument()
+    _ns::String = ""
+end
+
+
+
+#=
+
+Set of XPaths for variables in DDI (.xml) files
+
+=#
+
+EXTRACT_CONDITIONS = "/x:codeBook/x:stdyDscr/x:dataAccs/x:useStmt/x:conditions"
+EXTRACT_CITATION = "/x:codeBook/x:stdyDscr/x:dataAccs/x:useStmt/x:citReq"
+EXTRACT_IPUMS_PROJECT = "/x:codeBook/x:stdyDscr/x:citation/x:serStmt/x:serName"
+EXTRACT_NOTES = "/x:codeBook/x:stdyDscr/x:notes"
+EXTRACT_DATE = "/x:codeBook/x:stdyDscr/x:citation/x:prodStmt/x:prodDate/@date"
+
+VAR_NAME_XPATH = "/x:codeBook/x:dataDscr/x:var/@name"
+VAR_STARTPOS_XPATH = "/x:codeBook/x:dataDscr/x:var/x:location/@StartPos"
+VAR_ENDPOS_XPATH = "/x:codeBook/x:dataDscr/x:var/x:location/@EndPos"
+VAR_WIDTH_XPATH = "/x:codeBook/x:dataDscr/x:var/x:location/@width"
+VAR_LABL_XPATH = "/x:codeBook/x:dataDscr/x:var/x:labl"
+VAR_TXT_XPATH = "/x:codeBook/x:dataDscr/x:var/x:txt"
+VAR_DCML_XPATH = "/x:codeBook/x:dataDscr/x:var/@dcml"
+VAR_TYPE_XPATH = "/x:codeBook/x:dataDscr/x:var/x:varFormat/@type"
+VAR_INTERVAL_XPATH = "/x:codeBook/x:dataDscr/x:var/@intrvl"
+VAR_CATEGORY_XPATH = "/x:codeBook/x:dataDscr/x:var/x:catgry"
+
+
+
+"""
+    parse_ddi(filepath)
+
+Parses a valid IPUMs DDI XML file and returns a DDIInfo object.
+
+
+# Examples
+```julia-repl
+julia> parse_ddi("test_extract.xml")
+```
+"""
+function parse_ddi(filepath::String)
+
+    # check to make sure file is xml and not zip
+    
+    _check_that_file_is_xml(filepath)
+
+    # check to make sure file exists
+
+    _check_that_file_exists(filepath)
```

Comment:
I think this line may be redundant as I think Julia will throw a file does not exist anyways... I cannot recall